### PR TITLE
Deprecate rp.name

### LINKED ISSUE #2121: Deprecate rp.name
## Proposed Change

Remove, deprecate, or ignore `PublicKeyCredentialRPEntity.name`. It's not used by any credential provider that I know about.

This might be tricky because it's inherited from `PublicKeyCredentialEntity.name`.

### PR DESCRIPTION
**Pull request by emlun**
Closes #2121.

The following tasks have been completed:

- ~~[ ] Modified Web platform tests ([link](https://github.com/web-platform-tests/wpt/))~~ No change to client logic

Implementation commitment:

- ~~[ ] WebKit ([link to issue](https://bugs.webkit.org/))~~
- ~~[ ] Chromium ([link to issue](https://issues.chromium.org/issues/new?component=1456855&template=0))~~
- ~~[ ] Gecko ([link to issue](https://bugzilla.mozilla.org/home))~~

Documentation and checks

- ~~[ ] Affects privacy~~
- ~~[ ] Affects security~~
- ~~[ ] Updated explainer ([link](https://github.com/w3c/webauthn/wiki)~~

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/2159.html" title="Last updated on Oct 2, 2024, 7:10 PM UTC (fe68eae)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/2159/efdf948...fe68eae.html" title="Last updated on Oct 2, 2024, 7:10 PM UTC (fe68eae)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -3593,14 +3593,20 @@ associated with or [=scoped=] to, respectively.
     :   <dfn>name</dfn>
     ::  A [=human palatability|human-palatable=] name for the entity. Its function depends on what the {{PublicKeyCredentialEntity}} represents:
 
-          - When inherited by {{PublicKeyCredentialRpEntity}} it is a [=human palatability|human-palatable=] identifier for the [=[RP]=], intended only
+          - \[DEPRECATED] When inherited by {{PublicKeyCredentialRpEntity}} it is a [=human palatability|human-palatable=] identifier for the [=[RP]=], intended only
             for display. For example, "ACME Corporation", "Wonderful Widgets, Inc." or "ОАО Примертех".
 
+            This member is deprecated because many [=clients=] do not display it,
+            but it remains a required dictionary member for backwards compatibility.
+            [=[RPS]=] MAY as a safe default set this equal to the [=RP ID=].
```
Comment (agl) [inline index.bs:3601]:
```suggestion
            [=[RPS]=] MAY, as a safe default, set this equal to the [=RP ID=].
```

**Review by MasterKale — APPROVED** [review]

File: index.bs [hunk 2]
```diff
@@ -3593,14 +3593,20 @@ associated with or [=scoped=] to, respectively.
...
+
               - [=[RPS]=] SHOULD perform enforcement, as prescribed in Section 2.3 of
                 [[!RFC8266]] for the Nickname Profile of the PRECIS FreeformClass [[!RFC8264]],
                 when setting {{PublicKeyCredentialEntity/name}}'s value, or displaying the value to the user.
 
-              - This string MAY contain language and direction metadata. [=[RPS]=] SHOULD consider providing this information. See [[#sctn-strings-langdir]] about how this metadata is encoded.
+              - This string MAY contain language and direction metadata.
+                [=[RPS]=] SHOULD consider providing this information if setting the member to a value other than the [=RP ID=].
+                See [[#sctn-strings-langdir]] about how this metadata is encoded.
```
Comment (agl) [inline index.bs:3609]: Aren't we getting rid of this stuff? Wasn't that the outcome from TPAC?

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs]: Yeah, the problem with that is that [we implicitly recommend against empty string](https://github.com/w3c/webauthn/issues/2121#issuecomment-2289741275) by reference to the PRECIS profile, so omitting empty string here is a bit more consistent with that.

**Review by sbweeden — APPROVED** [review]
Looks good. I'm pretty sure most RPs are setting it to the RPID anyway.

**Comment by sbweeden** [comment]
Per discussion on WG call 2024-10-02, merging this as I was asked to be a secondary reviewer and there were no objections to merging after that review.

**Pull request merged**